### PR TITLE
feat: add predictive suggestions

### PR DESCRIPTION
**Pull request by lakhansamani**
## Proposed Changes
* Add prop called `enablePredictiveSuggestions` to React + Vue DataSearch components
* Add prop called `enablePredictiveSuggestions` to React CategorySearch component
* Default value for `enablePredictiveSuggestions` should be `false`

When set to true, it predicts the next relevant words from a field's value based on the search query typed by the user. When set to false (default), the entire field's value would be displayed. This may not be desirable for long-form fields (where average words per field value are greater than 4 and may not fit in a single line)

## How you have tested?

https://www.loom.com/share/a9188bcd19a5408c8539ef113bd5da1f

## Dependency

Need the following PR from reactivecore to be merged
https://github.com/appbaseio/reactivecore/pull/87

## StoryBook PRs
* Playground: https://github.com/appbaseio/playground/pull/63
* Vue Playground: https://github.com/appbaseio/vue-playground/pull/24

## Docs PR
* https://github.com/appbaseio/Docs/pull/166

**Before submitting a pull request,** please make sure the following is done:

- [x] Describe the proposed changes and how it'll improve the library experience.
- [x] Please make sure that there are no linting errors in the code.
- [x] Add a demo video/gif/screenshot to explain how did you test the fix.
- [ ] If it is a global change, try to add any side effects that it could have.
- [x] Create a PR to add/update the docs (if needed) at [here](https://github.com/appbaseio/Docs).
- [x] Create a PR to add/update the storybook (if needed) at [here](https://github.com/appbaseio/playground).

**Learn more about contributing:** [Contributing guides](https://github.com/appbaseio/reactivesearch/blob/next/.github/CONTRIBUTING.md)

### CODE REVIEW

File: packages/vue/examples/data-search/src/App.vue [hunk 1]
```diff
@@ -10,6 +10,7 @@
 				componentId="BookSensor"
 				:dataField="['original_title', 'original_title.search']"
 				:URLParams="true"
+				:enablePredictiveSuggestions="true"
```
Comment (bietkul) [inline packages/vue/examples/data-search/src/App.vue]: @lakhansamani Can you revert that change?

File: packages/web/examples/CategorySearch/src/index.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ const Main = () => (
 					dataField={['original_title', 'original_title.search']}
 					categoryField="authors.keyword"
 					componentId="BookSensor"
+					enablePredictiveSuggestions
```
Comment (bietkul) [inline packages/web/examples/CategorySearch/src/index.js]: @lakhansamani Just revert this change.

File: packages/web/examples/DataSearch/src/index.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ const Main = () => (
 					dataField={['original_title', 'original_title.search']}
 					componentId="BookSensor"
 					URLParams
+					enablePredictiveSuggestions
```
Comment (bietkul) [inline packages/web/examples/DataSearch/src/index.js]: @lakhansamani Just revert this line

**Review by bietkul — CHANGES_REQUESTED** [review]
@lakhansamani PR looks good :+1:
Just some minor changes needed.

**Comment by lakhansamani** [comment]
@bietkul made the requested chagnes

**Review by bietkul — APPROVED** [review]
